test/unit3/test_int.py

```diff
@@ -214,9 +215,20 @@ def test_basic(self):
         self.assertEqual(int('1z141z5', 36), 4294967297)
 
     def test_underscores(self):
+        for lit in VALID_UNDERSCORE_LITERALS:
+            if any(ch in lit for ch in '.eEjJ'):
+                continue
+            self.assertEqual(int(lit, 0), int(lit.replace('_', ''), 0))
+
+        for lit in INVALID_UNDERSCORE_LITERALS:
+            if any(ch in lit for ch in '.eEjJ'):
+                continue
+            self.assertRaises(ValueError, int, lit, 0)
+
         # Additional test cases with bases != 0, only for the constructor:
-        # self.assertEqual(int("1_00", 3), 9)
-        # self.assertEqual(int("0_100"), 100)  # not valid as a literal!
+        self.assertEqual(int("1_00", 3), 9)
+        self.assertEqual(int("0_100"), 100)  # not valid as a literal!
+        # self.assertEqual(int(b"1_00"), 100)  # byte underscore
```

Comment:
happy to keep bytes outside of the scope of this pr

if you want it inside the scope of this pr - then...

in the constructors for `float/int` adjust the test for string e.g. 

https://github.com/skulpt/skulpt/blob/master/src/float.js#L32

https://github.com/skulpt/skulpt/blob/master/src/int.js#L60

becomes 
```
if (Sk.builtin.checkString(x) || Sk.builtin.checkBytes(x))
```

then also 
https://github.com/skulpt/skulpt/blob/f81be786f7c32d3d40ff991559f02ee6a0e30ae7/src/int.js#L66
needs to be
```javascript
 val = Sk.str2number(x.$jsstr())
```

(though an issue for this approach is the error messages won't be accurate for invalid byte strings)